alf/environments/parallel_environment_test.py

```diff
@@ -100,6 +104,77 @@ def test_step(self):
                          time_step2.observation.shape)
         env.close()
 
+    def test_non_blocking_reset_with_spare_envs(self):
+        num_envs = 2
+        sleep_time = 1.
+        self._set_default_specs()
+
+        def slow_env_load(env_name, env_id=0):
+            return SlowStartingEnvironment(
+                self.observation_spec,
+                self.action_spec,
+                env_id=env_id,
+                max_duration=1,
+                time_sleep=sleep_time,
+                use_tensor_time_step=False)
+
+        start_t = time.time()
+        env = alf.environments.utils.create_environment(
+            env_name="IgnoredName",
+            env_load_fn=slow_env_load,
+            num_parallel_environments=num_envs,
+            start_serially=False,
+            num_spare_envs=num_envs)
+        init_t = time.time()
+        init_time = init_t - start_t
+        self.assertLessEqual(
+            init_time,
+            sleep_time * 2 - 0.5,
+            msg=('Expected all processes to start together, '
+                 'got {} wait time').format(init_time))
+
+        time_step0 = env.reset()
+        assert torch.all(time_step0.step_type == ds.StepType.FIRST)
+        action_spec = env.action_spec()
+        action = torch.stack([action_spec.sample() for _ in range(num_envs)])
+        time_step1 = env.step(action)
+        self.assertEqual(time_step0.observation.shape,
+                         time_step1.observation.shape)
+        step1_t = time.time()
+        time_step2 = env.step(action)
+        reset_t = time.time()
+        reset_time = reset_t - step1_t
```

Comment:
I still don't get why this is called reset_time. What I see it is the time for `env.step()`, not reset.

Reply:
Oh, this step() internally calls reset.  Added comment.